server/api/initial_data.py

```diff
@@ -571,14 +571,20 @@ def _migrate_artifacts_batch(
         # project - copy as is
         new_artifact.project = artifact_metadata.get("project", None)
 
-        # key - the artifact's key, without iteration if it is attached to it
-        key = artifact_metadata.get("key", "")
-        new_artifact.key = key
-
         # iteration - the artifact's iteration
         iteration = artifact_metadata.get("iter", None)
         new_artifact.iteration = int(iteration) if iteration else 0
 
+        # key - the artifact's key, without iteration if it is attached to it
+        key = (
+            artifact.key
+            or artifact_dict.get("spec", {}).get("db_key")
+            or artifact_metadata.get("key", "")
```

Comment:
Is this really expected to be meaningful? Won't `artifact.key` always have a value, since it's the `key` from the DB table?

Reply:
I guess not, we will also have `artifact.key`